code/db_model.py

```diff
@@ -0,0 +1,122 @@
+    station_name = db.Column(db.String(50), nullable=False)
+    station_address = db.Column(db.String(100), nullable=False)
+    coords = db.Column(Geometry(geometry_type='POINT'),
+                               nullable=False)
+
+    def __repr__(self):
+        return (f"Fire Station Name = {self.station_name} \n"
+                f"Fire Station Address = {self.station_address} \n")
+
+
+class SFHospital(db.Model):
+    """Model the table of SF hospitals where ambulances transport patients."""
+
+    __tablename__ = 'hospitals'
+
+    hospital_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    hospital_name = db.Column(db.String(100), nullable=False)
+    hospital_address = db.Column(db.String(100), nullable=False)
+    coords = db.Column(Geometry(geometry_type='POINT'), nullable=False)
+
+    def __repr__(self):
+        return (f"Hospital Name = {self.hospital_name} \n"
+                f"Hospital Address = {self.hospital_address} \n")
+
+
+class MedicalCall(db.Model):
+    """Model the table of emergency calls requiring an ambulance."""
+
+    __tablename__ = 'medical_calls'
+
+    call_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    als_unit = db.Column(db.Boolean, nullable=False)
+    coords = db.Column(Geometry(geometry_type='POINT'), nullable=False)
+    zipcode = db.Column(db.String(10))
+    call_type_group = db.Column(db.String(50))
+    neighborhood = db.Column(db.String(50), nullable=False)
+
+    vars = ['call_number', 'incident_number',
+            'number_alarms', 'unit_seq_dispatch']
+    var_dict = dict([(var, db.Column(db.Integer, nullable=False))
+                     for var in vars])
+
+    vars = ['call_date', 'watch_date']
+    var_dict.update([(var, db.Column(db.Date, nullable=False)) for var in vars])
+
+    vars = ['received_dttm', 'entry_dttm', 'dispatch_dttm']
+    var_dict.update([(var, db.Column(db.DateTime, nullable=False))
+                     for var in vars])
+
+    vars = ['response_dttm', 'onscene_dttm', 'transport_dttm',
+            'hospital_dttm', 'available_dttm']
+    var_dict.update([(var, db.Column(db.DateTime)) for var in vars])
+
+    vars = ['call_type', 'rowid', 'unit_type']
+    var_dict.update([(var, db.Column(db.String(25), nullable=False))
+                     for var in vars])
+
+    vars = ['address', 'location', 'call_final_disposition']
+    var_dict.update([(var, db.Column(db.String(100), nullable=False))
+                     for var in vars])
+
+    vars = ['city', 'tract', 'box']
+    var_dict.update([(var, db.Column(db.String(25))) for var in vars])
+
+    vars = ['battalion', 'final_priority',
+            'fire_district', 'supervisor_district']
+    var_dict.update([(var, db.Column(db.String(5), nullable=False))
+                     for var in vars])
+
+    vars = ['station_area', 'original_priority', 'priority']
+    var_dict.update([(var, db.Column(db.String(5))) for var in vars])
+
+    locals().update(var_dict)
```

Comment:
It converts the keys in `var_dict` from strings to variable names. It is equivalent to writing, for example, `station_area = db.Column(db.String(5))`. I chose to do it that way to avoid writing almost the same thing multiple times, e.g.

```
station_area = db.Column(db.String(5))
original_priority = db.Column(db.String(5))
priority = db.Column(db.String(5))
```

Would there be a better way to do it?

Reply:
ok I see, thanks for the explanation. I recommend just writing out all the columns for a few reasons:
- it follows the sqlalchemy convention and is therefore more readable. note that in a production codebase the code is written once, but read hundreds or thousands of times, so it's more important to optimize for readability than number of characters saved
- additionally, I'm not sure this `locals()` approach is saving that many characters considering you still need to list out all the variable names